providers/papermill/src/airflow/providers/papermill/operators/papermill.py

```diff
@@ -24,17 +24,16 @@
 import attr
 import papermill as pm
 
-from airflow.models import BaseOperator
 from airflow.providers.common.compat.lineage.entities import File
 from airflow.providers.common.compat.version_compat import AIRFLOW_V_3_0_PLUS
 from airflow.providers.papermill.hooks.kernel import REMOTE_KERNEL_ENGINE, KernelHook
+from airflow.providers.papermill.version_compat import BaseOperator
 
 if TYPE_CHECKING:
-    try:
+    if AIRFLOW_V_3_0_PLUS:
         from airflow.sdk.definitions.context import Context
-    except ImportError:
-        # TODO: Remove once provider drops support for Airflow 2
-        from airflow.utils.context import Context
+    else:
+        from airflow.utils.context import Context  # type: ignore[no-redef]
```

Comment:
Technically the try / catch here is right. 

I think we should enforce a general pattern over here, the 3.0 / 3.1 check was relevant for BaseOperator mainly due to a 3.1 xcom bug in decorators.